File: java-kata/src/test/java/com/leetcode/BinaryTreeLevelOrderTraversalTest.java
package com.leetcode;

import com.leetcode.data.TreeNode;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.assertj.core.api.Assertions.assertThat;

class BinaryTreeLevelOrderTraversalTest {

    private final BinaryTreeLevelOrderTraversal target = new BinaryTreeLevelOrderTraversal();

    @Test
    void test1() {
//        Input: root = [3,9,20,null,null,15,7]
//        Output: [[3],[9,20],[15,7]]

        var input = new TreeNode(3,
                new TreeNode(9),
                new TreeNode(20,
                        new TreeNode(15),
                        new TreeNode(7)
                )
        );
        assertThat(target.levelOrder(input))
                .containsExactly(
                        List.of(3),
                        List.of(9, 20),
                        List.of(15, 7)
                );
    }

    @Test
    void test2() {
//        Input: root = [1]
//        Output: [[1]]

        var input = new TreeNode(1);
        assertThat(target.levelOrder(input))
                .containsExactly(List.of(1));
    }

    @Test
    void test3() {
//        Input: root = []
//        Output: []

        assertThat(target.levelOrder(null))
                .isEmpty();
    }
}

File: kotlin-kata/src/test/kotlin/com/leetcode/BinaryTreeLevelOrderTraversalTest.kt
package com.leetcode

import com.leetcode.data.TreeNode
import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

internal class BinaryTreeLevelOrderTraversalTest {

    private val target: BinaryTreeLevelOrderTraversal = BinaryTreeLevelOrderTraversal()

    @Test
    fun test1() {
//        Input: root = [3,9,20,null,null,15,7]
//        Output: [[3],[9,20],[15,7]]

        val input = TreeNode(
            3,
            TreeNode(9),
            TreeNode(
                20,
                TreeNode(15),
                TreeNode(7)
            )
        )
        assertThat(target.levelOrder(input))
            .containsExactly(
                listOf(3),
                listOf(9, 20),
                listOf(15, 7)
            )
    }

    @Test
    fun test2() {
//        Input: root = [1]
//        Output: [[1]]

        val input = TreeNode(1)
        assertThat(target.levelOrder(input))
            .containsExactly(listOf(1))
    }

    @Test
    fun test3() {
//        Input: root = []
//        Output: []

        assertThat(target.levelOrder(null))
            .isEmpty()
    }
}